rest_framework/serializers.py

```diff
@@ -1002,8 +1003,16 @@ def get_fields(self):
                     field_cls = self._get_nested_class(depth, relation_info)
                     kwargs = get_nested_relation_kwargs(relation_info)
                 else:
-                    field_cls = self._related_class
                     kwargs = get_relation_kwargs(field_name, relation_info)
+                    to_field = kwargs.pop('to_field', None)
+                    # it seems that some tests/django initializers are setting
+                    # to_field where it is totally unnecessary
+                    if to_field and to_field != 'id':
```

Comment:
Can you add some clarification on this? The comment is a bit vauge about what the issue actually is, though I assume it has to do with `AutoField` considering it involves the `id` field?